<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Imshow</class>
 <widget class="QMainWindow" name="Imshow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>843</width>
    <height>627</height>
   </rect>
  </property>
  <property name="maximumSize">
   <size>
    <width>1000</width>
    <height>1000</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Imshow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>-10</y>
      <width>381</width>
      <height>421</height>
     </rect>
    </property>
    <property name="maximumSize">
     <size>
      <width>1000</width>
      <height>1000</height>
     </size>
    </property>
    <property name="text">
     <string>                                               Loading...</string>
    </property>
   </widget>
   <widget class="QPushButton" name="Format">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>40</y>
      <width>99</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>format</string>
    </property>
   </widget>
   <widget class="QFrame" name="frame">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>80</y>
      <width>221</width>
      <height>224</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <layout class="QFormLayout" name="formLayout">
     <property name="fieldGrowthPolicy">
      <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
     </property>
     <item row="0" column="0">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>H:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QSlider" name="h_max">
       <property name="maximum">
        <number>180</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>S:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QSlider" name="s_min">
       <property name="maximum">
        <number>255</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QSlider" name="s_max">
       <property name="maximum">
        <number>255</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>V:</string>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QSlider" name="v_min">
       <property name="maximum">
        <number>255</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="5" column="1">
      <widget class="QSlider" name="v_max">
       <property name="maximum">
        <number>255</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QSlider" name="h_min">
       <property name="maximum">
        <number>180</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="switch_hsv">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>40</y>
      <width>99</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>HSV</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="coordinates_show">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>10</y>
      <width>111</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Coordinates</string>
    </property>
   </widget>
   <widget class="QSplitter" name="splitter">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>440</y>
      <width>399</width>
      <height>113</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <widget class="QFrame" name="frame_2">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="1">
       <widget class="QSpinBox" name="hmin">
        <property name="maximum">
         <number>180</number>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string>~</string>
        </property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QSpinBox" name="hmax">
        <property name="maximum">
         <number>180</number>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>S:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QLabel" name="label_10">
        <property name="text">
         <string>~</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>V:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="smin">
        <property name="maximum">
         <number>255</number>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>~</string>
        </property>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="QSpinBox" name="smax">
        <property name="maximum">
         <number>255</number>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QSpinBox" name="vmin">
        <property name="maximum">
         <number>255</number>
        </property>
       </widget>
      </item>
      <item row="2" column="3">
       <widget class="QSpinBox" name="vmax">
        <property name="maximum">
         <number>255</number>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>H:</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QFrame" name="frame_3">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0" colspan="2">
       <widget class="QPushButton" name="automatic_threshold">
        <property name="text">
         <string>Automatic</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_11">
        <property name="text">
         <string>X:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="x_point">
        <property name="maximum">
         <number>381</number>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_12">
        <property name="text">
         <string>Y:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QSpinBox" name="y_point">
        <property name="maximum">
         <number>420</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QFrame" name="frame_4">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="0" column="0">
       <widget class="QLabel" name="h_name">
        <property name="text">
         <string>h:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="h_value">
        <property name="text">
         <string>0</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="s_name">
        <property name="text">
         <string>s:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="s_value">
        <property name="text">
         <string>0</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="v_name">
        <property name="text">
         <string>v:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLabel" name="v_value">
        <property name="text">
         <string>0</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_5">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>320</y>
      <width>221</width>
      <height>230</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <layout class="QGridLayout" name="gridLayout_5">
     <item row="0" column="0" colspan="2">
      <widget class="QLineEdit" name="lineEdit">
       <property name="text">
        <string>      Morphological operation</string>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="2" column="0" colspan="2">
      <widget class="QPushButton" name="reset">
       <property name="text">
        <string>reset</string>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QPushButton" name="dilate">
       <property name="text">
        <string>dilate</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QPushButton" name="erode">
       <property name="text">
        <string>erode</string>
       </property>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QPushButton" name="open">
       <property name="text">
        <string>open</string>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QPushButton" name="close">
       <property name="text">
        <string>close</string>
       </property>
      </widget>
     </item>
     <item row="5" column="0">
      <widget class="QPushButton" name="tophat">
       <property name="text">
        <string>tophat</string>
       </property>
      </widget>
     </item>
     <item row="5" column="1">
      <widget class="QPushButton" name="blackhat">
       <property name="text">
        <string>blackhat</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0" colspan="2">
      <widget class="QWidget" name="widget" native="true">
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="0" column="0">
         <widget class="QSpinBox" name="a">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>50</number>
          </property>
          <property name="value">
           <number>1</number>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLabel" name="X">
          <property name="text">
           <string>     X</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QSpinBox" name="b">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>50</number>
          </property>
          <property name="value">
           <number>1</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="Canny">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>10</y>
      <width>99</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>canny</string>
    </property>
   </widget>
   <widget class="QFrame" name="frame_6">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>140</y>
      <width>151</width>
      <height>103</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <layout class="QGridLayout" name="gridLayout_7">
     <item row="0" column="0">
      <widget class="QLabel" name="label_16">
       <property name="text">
        <string>blur:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1" colspan="2">
      <widget class="QSpinBox" name="blue_size">
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="value">
        <number>15</number>
       </property>
      </widget>
     </item>
     <item row="1" column="0" colspan="2">
      <widget class="QLabel" name="label_17">
       <property name="text">
        <string>hough:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QSpinBox" name="Gaussian_x">
       <property name="value">
        <number>15</number>
       </property>
      </widget>
     </item>
     <item row="2" column="2">
      <widget class="QSpinBox" name="Gaussian_y"/>
     </item>
    </layout>
   </widget>
   <widget class="QFrame" name="frame_7">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>10</y>
      <width>171</width>
      <height>117</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <layout class="QGridLayout" name="gridLayout_8">
     <item row="0" column="0" colspan="2">
      <widget class="QSpinBox" name="threshold_a">
       <property name="maximum">
        <number>255</number>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QLabel" name="label_13">
       <property name="text">
        <string> X</string>
       </property>
      </widget>
     </item>
     <item row="0" column="3">
      <widget class="QSpinBox" name="threshold_b">
       <property name="maximum">
        <number>255</number>
       </property>
       <property name="value">
        <number>255</number>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_14">
       <property name="text">
        <string>thresh_a</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1" colspan="3">
      <widget class="QSlider" name="Slider_thresha">
       <property name="maximum">
        <number>255</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="2" column="0" colspan="2">
      <widget class="QLabel" name="label_15">
       <property name="text">
        <string>thresh_b</string>
       </property>
      </widget>
     </item>
     <item row="2" column="2" colspan="2">
      <widget class="QSlider" name="Slider_threshb">
       <property name="maximum">
        <number>255</number>
       </property>
       <property name="value">
        <number>255</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QFrame" name="frame_8">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>260</y>
      <width>171</width>
      <height>80</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <layout class="QGridLayout" name="gridLayout_6">
     <item row="0" column="0">
      <widget class="QLabel" name="label_18">
       <property name="text">
        <string>ip :</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="lineEditIp">
       <property name="text">
        <string>192.168.1.1</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0" colspan="2">
      <widget class="QPushButton" name="connect">
       <property name="text">
        <string>connect</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>843</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuOpen">
    <property name="title">
     <string>open</string>
    </property>
    <addaction name="separator"/>
    <addaction name="Open_Creame"/>
    <addaction name="Open_Video"/>
    <addaction name="Open_Photo"/>
   </widget>
   <widget class="QMenu" name="Rect">
    <property name="title">
     <string>drawtarget</string>
    </property>
    <addaction name="actionRectangle"/>
    <addaction name="actionMin_rectangle"/>
    <addaction name="actionCenter"/>
    <addaction name="actionEllipse"/>
    <addaction name="actionConvexHull"/>
    <addaction name="actionClose"/>
   </widget>
   <widget class="QMenu" name="menuThreshold">
    <property name="title">
     <string>threshold</string>
    </property>
    <addaction name="actionBINARY"/>
    <addaction name="actionBINARY_INV"/>
    <addaction name="actionTRUNC"/>
    <addaction name="actionTOZERO"/>
    <addaction name="actionTOZERO_INV"/>
    <addaction name="separator"/>
    <addaction name="actionOTSU"/>
    <addaction name="actionTRIANGLE"/>
    <addaction name="separator"/>
    <addaction name="actionGAUSSIAN_C"/>
    <addaction name="actionMEAN_C"/>
   </widget>
   <widget class="QMenu" name="menuBlur">
    <property name="title">
     <string>blur</string>
    </property>
    <addaction name="actionBlur"/>
    <addaction name="actionGaussian"/>
    <addaction name="actionMedian"/>
    <addaction name="actionBilateral"/>
    <addaction name="actionMeanShift"/>
    <addaction name="actionEdgePreservingFilter"/>
   </widget>
   <widget class="QMenu" name="menuHough">
    <property name="title">
     <string>hough</string>
    </property>
    <addaction name="actionLines"/>
    <addaction name="actionLinesp"/>
    <addaction name="actionCircles"/>
   </widget>
   <addaction name="menuOpen"/>
   <addaction name="Rect"/>
   <addaction name="menuThreshold"/>
   <addaction name="menuBlur"/>
   <addaction name="menuHough"/>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <action name="Open_Creame">
   <property name="text">
    <string>Open Creame</string>
   </property>
  </action>
  <action name="Open_Video">
   <property name="text">
    <string>Open Video</string>
   </property>
  </action>
  <action name="Open_Photo">
   <property name="text">
    <string>Open Photo</string>
   </property>
  </action>
  <action name="actionRGB">
   <property name="text">
    <string>RGB</string>
   </property>
  </action>
  <action name="actionGRB">
   <property name="text">
    <string>GRB</string>
   </property>
  </action>
  <action name="actionHSV">
   <property name="text">
    <string>HSV</string>
   </property>
  </action>
  <action name="actionGRAY">
   <property name="text">
    <string>GRAY</string>
   </property>
  </action>
  <action name="actionRectangle">
   <property name="text">
    <string>max_rectangle</string>
   </property>
  </action>
  <action name="actionMin_rectangle">
   <property name="text">
    <string>min_rectangle</string>
   </property>
  </action>
  <action name="actionCenter">
   <property name="text">
    <string>center</string>
   </property>
  </action>
  <action name="actionClose">
   <property name="text">
    <string>close</string>
   </property>
  </action>
  <action name="actionEllipse">
   <property name="text">
    <string>ellipse</string>
   </property>
  </action>
  <action name="actionConvexHull">
   <property name="text">
    <string>convexHull</string>
   </property>
  </action>
  <action name="actionBINARY">
   <property name="text">
    <string>BINARY</string>
   </property>
  </action>
  <action name="actionBINARY_INV">
   <property name="text">
    <string>BINARY_INV</string>
   </property>
  </action>
  <action name="actionTRUNC">
   <property name="text">
    <string>TRUNC</string>
   </property>
  </action>
  <action name="actionTOZERO">
   <property name="text">
    <string>TOZERO</string>
   </property>
  </action>
  <action name="actionTOZERO_INV">
   <property name="text">
    <string>TOZERO_INV</string>
   </property>
  </action>
  <action name="actionOTSU">
   <property name="text">
    <string>OTSU</string>
   </property>
  </action>
  <action name="actionTRIANGLE">
   <property name="text">
    <string>TRIANGLE</string>
   </property>
  </action>
  <action name="actionGAUSSIAN_C">
   <property name="text">
    <string>GAUSSIAN_C</string>
   </property>
  </action>
  <action name="actionMEAN_C">
   <property name="text">
    <string>MEAN_C</string>
   </property>
  </action>
  <action name="actionBlur">
   <property name="text">
    <string>blur</string>
   </property>
  </action>
  <action name="actionGaussian">
   <property name="text">
    <string>gaussian</string>
   </property>
  </action>
  <action name="actionMedian">
   <property name="text">
    <string>median</string>
   </property>
  </action>
  <action name="actionMeanShift">
   <property name="text">
    <string>meanShift</string>
   </property>
  </action>
  <action name="actionEdgePreservingFilter">
   <property name="text">
    <string>edgePreservingFilter</string>
   </property>
  </action>
  <action name="actionBilateral">
   <property name="text">
    <string>bilateral</string>
   </property>
  </action>
  <action name="actionLines">
   <property name="text">
    <string>lines</string>
   </property>
  </action>
  <action name="actionLinesp">
   <property name="text">
    <string>linesp</string>
   </property>
  </action>
  <action name="actionCircles">
   <property name="text">
    <string>circles</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
